runtime/src/bank.rs

```diff
@@ -1358,20 +1358,20 @@ impl Bank {
     /// Hash the `accounts` HashMap. This represents a validator's interpretation
     ///  of the delta of the ledger since the last vote and up to now
     fn hash_internal_state(&self) -> Hash {
-        // If there are no accounts, return the same hash as we did before
-        // checkpointing.
-        if let Some(accounts_delta_hash) = self.rc.accounts.hash_internal_state(self.slot()) {
-            let mut signature_count_buf = [0u8; 8];
-            LittleEndian::write_u64(&mut signature_count_buf[..], self.signature_count() as u64);
-
-            hashv(&[
-                &self.parent_hash.as_ref(),
-                &accounts_delta_hash.as_ref(),
-                &signature_count_buf,
-            ])
-        } else {
-            self.parent_hash
-        }
+        // If there are no accounts, return the hash of the previous state and the latest blockhash
+        let accounts_delta_hash = self
+            .rc
+            .accounts
+            .hash_internal_state(self.slot())
+            .expect("No accounts delta was found for this bank, that should not be possible");
```

Comment:
how come we don't just change the return type of hash_internal_state()?

Reply:
Yeah that's fair. It shouldn't make a functional difference though, we ought to panic either way. want me to move it?

Reply:
yeah, I think so.  API semantics are that `accounts` is keeping all this together (via `new_from_parent()`), the `expect("")`s can move down one level